test/gtest/intervals.cpp

```diff
@@ -0,0 +1,121 @@
+#include "support/intervals.h"
+#include "ir/utils.h"
+#include "gtest/gtest.h"
+
+using namespace wasm;
+
+TEST(IntervalsTest, TestEmpty) {
+  std::vector<Interval> intervals;
+  std::vector<int> expected;
+  ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals), expected);
+}
+
+TEST(IntervalsTest, TestOne) {
+  std::vector<Interval> intervals;
+  intervals.emplace_back(Interval{0, 2, 5});
+  std::vector<int> expected{0};
+  ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals), expected);
+}
+
+TEST(IntervalsTest, TestNoOverlapFound) {
+  std::vector<Interval> intervals;
+  intervals.emplace_back(Interval{0, 4, 2});
+  intervals.emplace_back(Interval{4, 8, 2});
+  std::vector<int> expected{0, 1};
+  ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals), expected);
+}
+
+TEST(IntervalsTest, TestOverlapFound) {
+  std::vector<Interval> intervals;
+  intervals.emplace_back(Interval{0, 2, 5});
+  intervals.emplace_back(Interval{1, 4, 10});
+  intervals.emplace_back(Interval{4, 5, 15});
+  std::vector<int> expected{1, 2};
+  ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals), expected);
+}
+
+TEST(IntervalsTest, TestOverlapFoundA) {
```

Comment:
It would be best to give these more descriptive names like `TestOverlapInnerNested`, `TestOverlapOuterNested`, `TestOverlapAscendingSequence`, `TestOverlapDescendingSequence`, etc. Ideally the role each interval plays in a test should be clear from the name of the test.

Reply:
done